src/System.CodeDom/src/System/CodeDom/Compiler/CodeGenerator.cs

```diff
@@ -789,23 +812,48 @@ protected virtual void GenerateCompileUnit(CodeCompileUnit e)
 
         protected virtual void GenerateNamespace(CodeNamespace e)
         {
-            GenerateCommentStatements(e.Comments);
+            if (e == null)
+            {
+                throw new ArgumentNullException(nameof(e));
+            }
+
+            if (e.Comments.Count > 0)
+            {
+                GenerateCommentStatements(e.Comments);
+            }
```

Comment:
Ditto.  I'll stop commenting on these; it's not clear to me why we need to add these Count checks, as all of the called methods handle empty collections just fine.